### PR TITLE
Turn "Rename Discussion" dialog into a modal, closes #616

### LINKED ISSUE #616: Turn "Rename Discussion" dialog into a modal
Just what the title says. Something that matches the Edit Tags modal, etc. would be snazzier.

### PR DESCRIPTION
**Pull request by datitisev**
Changed "Rename Discussion" dialog into a modal. This PR closes #616. No language files yet, just want to get your feedback.
- Added `DiscussionRenameModal` component (Modal)
- Changed `DiscussionControls.renameAction` to use the modal (I may have removed the ability to return a promise)

``` js
/**
 * Rename the discussion.
 *
 * @return {Promise}
 */
renameAction() {
  const discussion = this;
  return app.modal.show(new DiscussionRenameModal({
    currentTitle: discussion.title(),
    discussion: discussion
  }));
}
```

I still have the `return` in the `onsubmit` modal event, but I don't think there's any way to make `renameAction` return a Promise.

I chose `core.forum.discussion_controls.rename_modal.title` and `core.forum.discussion_controls.rename_modal.submit_button` because Dominion said to keep things organized, but we can probably still use `core.forum.discussion_controls.rename_text` and `core.ref.save_changes`.

captura de pantalla 2016-07-26 a las 5 44 26 pm

---

---

_Original #1010 on July 26, 2016. Made a new PR to fix conflicts with forum dist_

### CODE REVIEW

**Comment by Cu8eeeR** [comment]
Are there more dialogs which need to be turned into modals? I havent checked them

**Comment by datitisev** [comment]
@Cu8eeeR After a quick search with the term `alert` through flarum/core, turns out there is an alert in the [Appearance Page](https://github.com/flarum/core/blob/1ebb8bf39ac6f401fa065be5776ab6dcc1d78506/js/admin/src/components/AppearancePage.js#L106).
I may make a new PR for it if I have the time to change that one as well.

**Comment by AmauryCarrade** [comment]
You forgot to check for `confirm`, and there are five of them:
- to [confirm the deletion of a group](https://github.com/flarum/core/blob/34f3d93ce56ad18ada6aad65bfc77244fea09716/js/admin/src/components/EditGroupModal.js#L97) in the permissions page;
- to [confirm the deletion of an user](https://github.com/flarum/core/blob/34f3d93ce56ad18ada6aad65bfc77244fea09716/js/forum/src/utils/UserControls.js#L98);
- to [confirm the composer can be closed](https://github.com/flarum/core/blob/34f3d93ce56ad18ada6aad65bfc77244fea09716/js/forum/src/components/Composer.js#L260) with text wrote in it;
- to [confirm a discussion can be deleted permanently](https://github.com/flarum/core/blob/34f3d93ce56ad18ada6aad65bfc77244fea09716/js/forum/src/components/Composer.js#L260);
- and finally to [confirm that all discussions can be marked as read](https://github.com/flarum/core/blob/34f3d93ce56ad18ada6aad65bfc77244fea09716/js/forum/src/components/IndexPage.js#L367) in the index page.

**Comment by datitisev** [comment]
@Cu8eeeR True! Forgot about those 🙄 . I may make a PR soon to fix those 😃 
Thanks for reporting! Maybe you should transform your comment into an issue, though. Just a thought ^_^

**Comment by datitisev** [comment]
Well, I haven't found any 'Alert' function to show an alert as a modal without having to create a class extending [`Alert`](https://datitisev.github.io/FlarumAPIDocs/js/master/class/js/lib/components/Alert.js~Alert.html)... and that would mean that we need tons of classes just for a confirmation 🤔 
Just wish for the new API changes that include this 😜 

I've tried to do `new Alert.component({ controls, type })`, but it gives me an error saying `this.initProps` is not defined in the actual `component` method :/

**Comment by franzliedke** [comment]
@datitisev Thanks for your PR!

Can you rebase (or at least recompile) again, please? Currently the diff shows that you are removing punycode.js from the dist file.

**Comment by datitisev** [comment]
@franzliedke Done! The only reason you see the first commit down here is because I rebased my branch to include all the new changes (I can't remember why now).

**Comment by Luceos** [comment]
@datitisev are all strings translated now, if not can you either provide a list so we can create a follow-up issue or can you translate them to English if you feel confident to pick that up?

**Comment by franzliedke** [comment]
@Luceos Because of the language file restructuring, I'd prefer if we could create a ticket for this, and fix that once the next-* branches are merged.

**Comment by Luceos** [comment]
@franzliedke sure, if @datitisev would be so kind either to provide a list here or create a new issue and relate back, that would be fantastic.

**Comment by Luceos** [comment]
Never mind, already did it. @franzliedke feel free to merge when ready.

File: js/forum/src/utils/DiscussionControls.js [hunk 1]
```diff
@@ -227,19 +228,10 @@ export default {
...
-    // indicating that the discussion was renamed.
-    if (title && title !== currentTitle) {
-      return this.save({title}).then(() => {
-        if (app.viewingDiscussion(this)) {
-          app.current.stream.update();
-        }
-        m.redraw();
-      });
-    }
+    const discussion = this;
```
Comment (franzliedke) [inline js/forum/src/utils/DiscussionControls.js]: This variable shouldn't be necessary, right?

File: js/forum/src/components/DiscussionRenameModal.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      </div>
+    )
+  }
+
+  onsubmit(e) {
+    e.preventDefault();
+
+    this.loading = true;
+
+    // const title = $('.DiscussionRenameModal input.title').val();
```
Comment (franzliedke) [inline js/forum/src/components/DiscussionRenameModal.js]: Please remove this line.

File: js/forum/src/components/DiscussionRenameModal.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ */
+export default class DiscussionRenameModal extends Modal {
+  init() {
+    super.init();
+
+    this.discussion = this.props.discussion;
+    this.currentTitle = this.props.currentTitle;
+    this.newTitle = m.prop(this.currentTitle);
+
+  }
```
Comment (franzliedke) [inline js/forum/src/components/DiscussionRenameModal.js]: While you're at it, please remove the empty line. :)

File: js/forum/src/components/DiscussionRenameModal.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  className() {
+    return 'DiscussionRenameModal Modal--small';
+  }
+
+  title() {
+    return app.translator.trans('core.forum.discussion_controls.rename_modal.title');
+  }
+
+  content() {
+
```
Comment (franzliedke) [inline js/forum/src/components/DiscussionRenameModal.js]: And this one. :)

**Comment by datitisev** [comment]
@franzliedke Done! :)

**Comment by franzliedke** [comment]
Thanks again, @datitisev. Great job!

**Pull request merged**